src/Services/DeviceService.php

```diff
@@ -0,0 +1,100 @@
+ * @package openemr
+ * @link      http://www.open-emr.org
+ * @author    Stephen Nielson <stephen@nielson.org>
+ * @copyright Copyright (c) 2021 Stephen Nielson <stephen@nielson.org>
+ * @license   https://github.com/openemr/openemr/blob/master/LICENSE GNU General Public License 3
+ */
+
+namespace OpenEMR\Services;
+
+use OpenEMR\Common\Database\QueryUtils;
+use OpenEMR\Common\Logging\SystemLogger;
+use OpenEMR\Common\Uuid\UuidRegistry;
+use OpenEMR\Services\Search\FhirSearchWhereClauseBuilder;
+use OpenEMR\Validators\ProcessingResult;
+
+class DeviceService extends BaseService
+{
+    private const DEVICE_TABLE = "lists";
+    private $uuidRegistry;
+
+    public function __construct()
+    {
+        parent::__construct('lists');
+        $this->uuidRegistry = new UuidRegistry(['table_name' => self::DEVICE_TABLE]);
+        $this->uuidRegistry->createMissingUuids();
+    }
+
+    public function search($search, $isAndCondition = true)
+    {
+        $sql = "
+            select l.*
+            , patients.*
+            from
+            (
+                SELECT 
+                    `udi`,
+                    `uuid`, `date`, `title`,`udi_data`, `begdate`, `diagnosis`, `user`, `pid`
+                FROM lists WHERE `type` = 'medical_device'
+            ) l
+            JOIN (
+                SELECT `pid`,`uuid` AS `puuid`
+                from patient_data
+            ) patients ON l.pid = patients.pid";
+
+        $search = is_array($search) ? $search : [];
+
+        $whereClause = FhirSearchWhereClauseBuilder::build($search, $isAndCondition);
+
+        $sql .= $whereClause->getFragment();
+        $sqlBindArray = $whereClause->getBoundValues();
+        $statementResults =  QueryUtils::sqlStatementThrowException($sql, $sqlBindArray);
+
+        $processingResult = new ProcessingResult();
+        while ($row = QueryUtils::fetchArrayFromResultSet($statementResults)) {
+            $record = $this->createResultRecordFromDatabaseResult($row);
+            $processingResult->addData($record);
+        }
+        return $processingResult;
+    }
+
+    public function getUuidFields(): array
+    {
+        return ['uuid', 'puuid'];
+    }
+
+    protected function createResultRecordFromDatabaseResult($row)
+    {
+        // handle any uuids
+        $record = parent::createResultRecordFromDatabaseResult($row);
+
+        $json = $record['udi_data'] ?? '{}';
+        if (!empty($record['diagnosis'])) {
+            $record['code'] = $this->addCoding($record['diagnosis']);
+            $record['code_full'] = $record['diagnosis'];
+        }
+       
+        try {
+            $dataSet = json_decode($json, JSON_THROW_ON_ERROR);
+            $standardElements = $dataSet['standard_elements'] ?? [];
+            unset($record['udi_data']); // don't send back the JSON array
+            $record['udi_di'] = $standardElements['di'] ?? null;
+            // TODO: @adunsulag check with @brady.miller should this be companyName, or issuingAgency?
```

Comment:
@bradymiller In your review can you tell me if this should be companyName for the device manufacturer in the JSON dump that we get back for the UDI, or whether using issuingAgency is the correct approach here.  I don't know enough about the UDI API / devices in general to determine this.

Reply:
@adunsulag , would use companyName
(Issuing agency is one of the 3 agencies that create/release/track the DI code)

Reply:
Thanks, I got this resolved now.